### ISSUE TITLE
Missing queryFn error in react-native

### ISSUE THREAD

**Issue by Carlosted** [tannerlinsley/react-query#2096]
**Describe the bug**
When I run a basic useQuery command in react native I get `missing queryFn` error

**To Reproduce**

use react-native SDK 40 with react-query v 3.13.4

My code:
```javascript
import React from "react";
import { useQuery } from "react-query";
import { View, Text } from "react-native";

export default function Component() {
  const fetchData = async () => {
    const res = await fetch("https://jsonplaceholder.typicode.com/posts/1");
    return res.json();
  };

  const {data, isLoading, error} = useQuery("posts", fetchData);

  console.log(data, isLoading, error);

if (isLoading) {
    return <View><Text>loading...</Text></View>
}

  return (
    <View>
        <Text>{data}</Text>
    </View>
  );
}

```

**Expected behavior**
should see the data appearing on screen

**Screenshots**
<img width="887" alt="Screen Shot 2021-04-07 at 11 33 53 AM" src="https://user-images.githubusercontent.com/76854355/113893787-56f8cf80-9795-11eb-8da0-b7585e9c4380.png">

**Desktop**
 - OS: macOS 11.2 (Apple M1)
 - Browser: Chrome

**Smartphone Simulator**
 - Device: iPhone 12 pro max
 - OS: iOS 14.4
 - Browser: Expo Go
 - Expo Version: SDK 40

**Comment by Josh865** [comment]
I'm not sure if this is causing your issue, but I believe you need to await res.json() in your query function:

```
const fetchData = async () => {
    const res = await fetch("https://jsonplaceholder.typicode.com/posts/1");
    const data = await res.json();

    return data;
};
```

You would then need to output a property of the response rather than the response object:

```
return (
    <View>
        <Text>{data.body}</Text>
    </View>
);
```

**Comment by krall12** [comment]
@Carlosted what was your solution to this? I'm seeing this warning using useQueries in react-native

**Comment by Carlosted** [comment]
> @Carlosted what was your solution to this? I'm seeing this warning using useQueries in react-native

To be honest, I don't remember exactly what I did but I switched to axios to fetch data and, if I remember correctly it may have had something to do with the request being cached (not 100% sure). I hope this can help.

**Comment by dan-lee** [comment]
We had tens of thousands of these errors piling up in our app and I couldn't really reproduce them locally. I just found the reason why:

We registered a headless task that periodically ran some tasks (with `react-native-background-fetch`). One of these tasks mistakenly contained code similar to this:

```tsx
await queryClient.fetchQuery('/some/key')
```

As the headless task runs in its own context, no previous query fetchers were registered and ran into the `Missing queryFn` error.
Usually a fetcher is registered in runtime with something like `useQuery('/some/key')`. After this, the `fetchQuery` would work.

Maybe this can help someone else figure out what's going on.
